### PR TITLE
Adding script to revoke set of contracts

### PR DESCRIPTION
**Pull request by lucasege**
### Description

Adds bash script that consumes a list of contracts to revoke, withdraw, and refund/destroy the contracts.

Script also checks for things like: that the contract exists, it has created an account, the contract has unlocked all of its gold, and that the contract's gold has passed the unlocking period.

Example contract file:
`contract_file.txt`:
```
0x34BCf674AD932711AE791BD17371395281B79C32
0x6bB206ffdDe4b7f7744B01B1437F57E621E8426c
```

### Tested

Tested on local testnets and ganache

### CODE REVIEW

File: packages/protocol/scripts/bash/revoke_contracts.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+fi
+
+for contract in $(<"$1"); do
+  echo "Contract $contract"
+
+  # Check that account has been created
+  accountExists=1
+  celocli lockedgold:show "$contract" > /dev/null 2>&1 || accountExists=0
+  if [ $accountExists -eq 0 ]; then
+    echo "Account not created, rejecting $contract"
```
Comment (asaj) [inline packages/protocol/scripts/bash/revoke_contracts.sh]: Do we actually need to reject here? If they didn't create an account that just means that they never locked any CELO, and so we *should* be able to skip straight to revoking, right?

File: packages/protocol/scripts/bash/revoke_contracts.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    rejects+=( $contract )
+    continue
+  fi
+
+  # Check that CELO has been unlocked
+  lockedgold=$(celocli lockedgold:show "$contract")
+  total=$(echo "$lockedgold" | grep -o "total: [0-9]*" | cut -f2 -d : | tr -d '[:space:]')
```
Comment (asaj) [inline packages/protocol/scripts/bash/revoke_contracts.sh]: Assuming this doesn't include pending withdrawals

File: packages/protocol/scripts/bash/revoke_contracts.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  if [ "$total" != "0" ]; then
+    echo "Reject $contract"
```
Comment (asaj) [inline packages/protocol/scripts/bash/revoke_contracts.sh]: nit: `echo "Reject $contract: locked CELO > 0"`

File: packages/protocol/scripts/bash/revoke_contracts.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  fi
+
+  contractExists=1
+  celocli releasegold:revoke --contract "$contract" --yesreally > /dev/null || contractExists=0
+  if [ "$contractExists" -eq 0 ]; then
+    echo "Contract doesn't exist, rejecting $contract"
+    rejects+=( $contract )
+    continue
+  fi
+  celocli releasegold:locked-gold --contract "$contract" --action withdraw --value "$total" --yes
```
Comment (asaj) [inline packages/protocol/scripts/bash/revoke_contracts.sh]: What happens here if there are no pending withdrawals?

**Review by asaj — COMMENTED** [review]
Just some minor comments and questions

File: packages/protocol/scripts/bash/revoke_contracts.sh (hunk 4 above)
Comment (lucasege) [reply packages/protocol/scripts/bash/revoke_contracts.sh]: It's a no-op if no pending withdrawals so this should be ok

File: packages/protocol/scripts/bash/revoke_contracts.sh (hunk 2 above)
Comment (lucasege) [reply packages/protocol/scripts/bash/revoke_contracts.sh]: Yep this is just to verify that there is no gold locked.

While the $total here is passed in to `releasegold:locked-gold.withdraw`, the `withdraw` fn actually just withdrawals all available pending withdrawals, regardless of value. FWIW, this is exactly how `lockedgold:withdraw` works, but I guess I forgot to make `value` optional for the `action.withdraw` on `releasegold:locked-gold`

This has no real impact on functionality, but is unclear, so should be more clear. I'll create a tech debt issue

**Review by marekolszewski — APPROVED** [review]
Approved

**Pull request merged**